### PR TITLE
Shift grid to driver

### PR DESCRIPTION
**Pull request by nfehn**
Similar to PR #596, now for the rest of ExaDG.

### CODE REVIEW

File: applications/structure/beam/application.h [hunk 1]
```diff
@@ -205,10 +207,7 @@ class Application : public ApplicationBase<dim, Number>
         if(dim == 3)
           repetitions[2] = this->repetitions2;
 
-        dealii::GridGenerator::subdivided_hyper_rectangle(*this->grid->triangulation,
```
Comment (nfehn) [inline applications/structure/beam/application.h:208]: Looks like the use of `this->grid->triangulation` inside `lambda_create_triangulation()` for this particular application is a bug in the current implementation!?

**Review by richardschu — COMMENTED** [review]
looks good, we discussed the implications in and around #596 and #586 :+1:

**Comment by nfehn** [comment]
I think I have now addressed all the modules.

File: include/exadg/incompressible_navier_stokes/precursor/driver.cpp [hunk 2]
```diff
@@ -108,8 +109,44 @@ Driver<dim, Number>::synchronize_time_step_size() const
...
+  solver_precursor.time_integrator->set_current_time_step_size(time_step_size);
+  solver_main.time_integrator->set_current_time_step_size(time_step_size);
+}
+
+// performs some additional parameter checks
+template<int dim, typename Number>
+void
+Driver<dim, Number>::consistency_checks() const
+{
+  AssertThrow(application->precursor->get_parameters().ale_formulation == false,
```
Comment (richardschu) [inline include/exadg/incompressible_navier_stokes/precursor/driver.cpp:121]: :+1:

File: include/exadg/poisson/overset_grids/user_interface/application_base.h [hunk 3]
```diff
@@ -277,54 +263,23 @@ class ApplicationBase
     AssertThrow(domain2.get(), dealii::ExcMessage("Domain 2 is uninitialized."));
 
     domain1->add_parameters(prm, {"Domain1"});
-    domain2->add_parameters(prm, {"Domain1"});
+    domain2->add_parameters(prm, {"Domain2"});
```
Comment (richardschu) [inline include/exadg/poisson/overset_grids/user_interface/application_base.h:266]: :eye:

File: include/exadg/poisson/overset_grids/driver.cpp [hunk 4]
```diff
@@ -50,11 +50,31 @@ Driver<dim, n_components, Number>::setup()
 {
   pcout << std::endl << "Setting up Poisson solver for overset grids:" << std::endl;
 
-  application->setup();
+  // setup application
+  AssertThrow(application->domain1.get(), dealii::ExcMessage("Domain 1 is uninitialized."));
```
Comment (richardschu) [inline include/exadg/poisson/overset_grids/driver.cpp:54]: this is the content of the deleted `application->setup` function, right? .. containing the `set_boundary_ids()`, e.g., which you now moved here directly? Deleting the definitions in `include/exadg/poisson/overset_grids/user_interface/application_base.h`.

**Review by richardschu — APPROVED** [review]
thanks for carrying over the idea from #596  . looks good!

File: include/exadg/poisson/overset_grids/user_interface/application_base.h (hunk 3 above)
Comment (richardschu) [reply include/exadg/poisson/overset_grids/user_interface/application_base.h:266]: was that a :bug: ?

**Pull request merged**